yt/python/yt/cli/yt_binary.py

```diff
@@ -1276,13 +1276,65 @@ def add_start_query_parser(add_parser):
     parser.add_argument("engine", type=str, help='engine of a query, one of "ql", "yql", "chyt", "spyt"')
     parser.add_argument("query", type=str, help="query text")
     add_structured_argument(parser, "--settings", help="additional settings of a query in structured form")
-    add_structured_argument(parser, "--files", help='query files, a YSON list of files, each of which is represented by a map with keys "name", "content", "type".'
+    add_structured_argument(parser, "--files", help='query files, a YSON list of files, each of which is represented by a map with keys "name", "content", "type". '
                                                     'Field "type" is one of "raw_inline_data", "url"')
     parser.add_argument("--access-control-object", type=str, help='optional access control object name (deprecated)')
     add_structured_argument(parser, "--access-control-objects", help='access control objects, a YSON list of ACO names')
     parser.add_argument("--stage", type=str, help='query tracker stage, defaults to "production"')
 
 
+@copy_docstring_from(yt.run_query)
+def run_query(*args, format=None, structured=None, **kwargs):
+    is_async = kwargs.pop("async", False)
+
+    if structured:
+        if format is None:
+            format = yt.format.create_format(OUTPUT_FORMATS[YT_STRUCTURED_DATA_FORMAT])
+    else:
+        # Perform an early check similar to the one in read_query_result to
+        # avoid late errors (after query execution) in case user forgets to specify format.
+        if format is None:
+            raise yt.YtError("You should specify format")
+
+    query = yt.run_query(*args, sync=not is_async, **kwargs)
+    if is_async:
+        print(query.id)
+        return
+
+    if structured:
+        result = []
+        for query_result in query.get_results():
+            attributes = query_result.get_attributes()
+            if "error" not in attributes:
+                attributes["rows"] = builtins.list(query_result.read_rows(validate_not_truncated=False, raw=False))
+            result.append(attributes)
+        print_to_output(format.dumps_node(result))
+    else:
+        query_results = query.get_results()
+        if len(query_results) == 1:
+            query_result = query_results[0]
+            write_silently(chunk_iter_stream(query_result.read_rows(format=format), yt.config["read_buffer_size"]))
```

Comment:
Why do we `validate_not_truncated` in case of structured format and not in this case?

Reply:
My reasoning was the following: default is structured = false, some people will use `yt query 'select * from ...'` as a replacement for `yt read-table` in cases when slight alteration of data is needed (or if people are more used to expressing their thoughts in SQL), so silently outputting the truncated data is definitely bad in that case.

The structured = false is more like a combination of a sequence of get_query_result + read_query_result, it is more like "provide me with everything we know about the query result". I think that people invoking this method are more ready to deal with the fact result is incomplete -- at the very least, the marker about it being incomplete will be present in the output, i.e. the information is not dropped.

Finally, there are examples of silently truncating structured commands (e.g. `yt list` by default just truncates long lists without throwing an error, and outputs a truncation marker only in case when yson format is requestsed).

So, such behavior is more or less consistent with the current CLI traditions and is sane in primary usecases.